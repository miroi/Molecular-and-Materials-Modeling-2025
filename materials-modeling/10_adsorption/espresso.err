--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 5 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
[pc7415:08914] 7 more processes have sent help message help-mpi-api.txt / mpi-abort
[pc7415:08914] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
