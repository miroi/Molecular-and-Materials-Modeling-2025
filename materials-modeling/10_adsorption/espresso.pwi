&CONTROL
   verbosity        = 'low'
   tstress          = .true.
   tprnfor          = .true.
   outdir           = './tmp'
   prefix           = 'c_hg'
   pseudo_dir       = './'
/
&SYSTEM
   ibrav            = 0
   nat              = 9
   ntyp             = 2
   ecutwfc          = 65
   occupations      = 'smearing'
   degauss          = 0.01
   smearing         = 'gauss'
   assume_isolated  = '2D'
/
&ELECTRONS
   conv_thr         = 1e-10
/
&IONS
/
&CELL
/
&FCP
/
&RISM
/
ATOMIC_SPECIES
C 12.011 C.upf
Hg 200.592 Hg.upf

K_POINTS automatic
1 1 1  0 0 0

CELL_PARAMETERS angstrom
4.91200017930000 0.00000000000000 0.00000000000000
-2.45600065610000 4.25391661160000 0.00000000000000
0.00000000000000 0.00000000000000 0.00000000000000

ATOMIC_POSITIONS angstrom
C 0.0000000000 0.0000000000 7.5000000000  
C -1.2280003280 2.1269583060 7.5000000000  
C 2.4560000900 0.0000000000 7.5000000000  
C 1.2279997620 2.1269583060 7.5000000000  
C 1.2279998770 0.7089860600 7.5000000000  
C -0.0000003780 2.8359442390 7.5000000000  
C 3.6839999670 0.7089860600 7.5000000000  
C 2.4559997120 2.8359442390 7.5000000000  
Hg 2.4559997120 2.8359442390 9.5000000000  

