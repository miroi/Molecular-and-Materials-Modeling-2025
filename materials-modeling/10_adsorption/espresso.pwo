
     Program PWSCF v.7.4.1 starts on 29Aug2025 at 15:36:24 

     This program is part of the open-source Quantum ESPRESSO suite
     for quantum simulation of materials; please cite
         "P. Giannozzi et al., J. Phys.:Condens. Matter 21 395502 (2009);
         "P. Giannozzi et al., J. Phys.:Condens. Matter 29 465901 (2017);
         "P. Giannozzi et al., J. Chem. Phys. 152 154105 (2020);
          URL http://www.quantum-espresso.org", 
     in publications or presentations arising from this work. More details at
     http://www.quantum-espresso.org/quote

     Parallel version (MPI), running on     8 processors

     MPI processes distributed on     1 nodes
     15203 MiB available memory on the printing compute node when the environment starts

     Reading input from espresso.pwi
Warning: card &CELL ignored
Warning: card / ignored
Warning: card &FCP ignored
Warning: card / ignored
Warning: card &RISM ignored
Warning: card / ignored

     Current dimensions of program PWSCF are:
     Max number of different atomic species (ntypx) = 10
     Max number of k-points (npk) =  40000
     Max angular momentum in pseudopotentials (lmaxx) =  4

 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
     Error in routine invmat (3):
     error in DGETRF
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

     stopping ...
