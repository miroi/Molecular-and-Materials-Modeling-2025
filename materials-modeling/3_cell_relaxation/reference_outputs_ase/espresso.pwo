
     Program PWSCF v.7.4.1 starts on 17Jul2025 at 14:56:48 

     This program is part of the open-source Quantum ESPRESSO suite
     for quantum simulation of materials; please cite
         "P. Giannozzi et al., J. Phys.:Condens. Matter 21 395502 (2009);
         "P. Giannozzi et al., J. Phys.:Condens. Matter 29 465901 (2017);
         "P. Giannozzi et al., J. Chem. Phys. 152 154105 (2020);
          URL http://www.quantum-espresso.org", 
     in publications or presentations arising from this work. More details at
     http://www.quantum-espresso.org/quote

     Parallel version (MPI), running on     4 processors

     MPI processes distributed on     1 nodes
     15166 MiB available memory on the printing compute node when the environment starts

     Reading input from espresso.pwi
Warning: card &CELL ignored
Warning: card / ignored
Warning: card &FCP ignored
Warning: card / ignored
Warning: card &RISM ignored
Warning: card / ignored

     Current dimensions of program PWSCF are:
     Max number of different atomic species (ntypx) = 10
     Max number of k-points (npk) =  40000
     Max angular momentum in pseudopotentials (lmaxx) =  4
     Message from routine setup:
     using ibrav=0 with symmetry is DISCOURAGED, use correct ibrav instead

     R & G space division:  proc/nbgrp/npool/nimage =       4
     Subspace diagonalization in iterative solution of the eigenvalue problem:
     a serial algorithm will be used


     Parallelization info
     --------------------
     sticks:   dense  smooth     PW     G-vecs:    dense   smooth      PW
     Min         238     238     73                 4892     4892     821
     Max         239     239     74                 4896     4896     823
     Sum         955     955    295                19579    19579    3287

     Using Slab Decomposition



     bravais-lattice index     =            0
     lattice parameter (alat)  =       7.3063  a.u.
     unit-cell volume          =     275.7930 (a.u.)^3
     number of atoms/cell      =            2
     number of atomic types    =            1
     number of electrons       =         8.00
     number of Kohn-Sham states=            4
     kinetic-energy cutoff     =      65.0000  Ry
     charge density cutoff     =     260.0000  Ry
     scf convergence threshold =      1.0E-08
     mixing beta               =       0.7000
     number of iterations used =            8  plain     mixing
     Exchange-correlation= PBE
                           (   1   4   3   4   0   0   0)

     celldm(1)=   7.306332  celldm(2)=   0.000000  celldm(3)=   0.000000
     celldm(4)=   0.000000  celldm(5)=   0.000000  celldm(6)=   0.000000

     crystal axes: (cart. coord. in units of alat)
               a(1) = (  -0.000000   0.707107   0.707107 )  
               a(2) = (   0.707107   0.000000   0.707107 )  
               a(3) = (   0.707107   0.707107   0.000000 )  

     reciprocal axes: (cart. coord. in units 2 pi/alat)
               b(1) = ( -0.707107  0.707107  0.707107 )  
               b(2) = (  0.707107 -0.707107  0.707107 )  
               b(3) = (  0.707107  0.707107 -0.707107 )  


     PseudoPot. # 1 for Si read from file:
     ./Si.upf
     MD5 check sum: 02fab3f35e82123ef5bf1cb05d5b1a5e
     Pseudo is Norm-conserving + core correction, Zval =  4.0
     Generated using ONCVPSP code by D. R. Hamann
     Using radial grid of 1510 points,  6 beta functions with: 
                l(1) =   0
                l(2) =   0
                l(3) =   1
                l(4) =   1
                l(5) =   2
                l(6) =   2

     atomic species   valence    mass     pseudopotential
     Si                4.00    28.08500     Si( 1.00)

     48 Sym. Ops., with inversion, found (24 have fractional translation)



   Cartesian axes

     site n.     atom                  positions (alat units)
         1        Si     tau(   1) = (   0.3535534   1.0606602   1.0606602  )
         2        Si     tau(   2) = (   0.0000000   0.0000000   0.0000000  )

     number of k points=   120

     Number of k-points >= 100: set verbosity='high' to print them.

     Dense  grid:    19579 G-vectors     FFT dimensions: (  40,  40,  40)

     Estimated max dynamical RAM per process >       9.42 MB

     Estimated total dynamical RAM >      37.69 MB

     Initial potential from superposition of free atoms

     starting charge       7.9988, renormalised to       8.0000
     Starting wfcs are    8 randomized atomic wfcs

     total cpu time spent up to now is        0.4 secs

     Self-consistent Calculation

     iteration #  1     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.00E-02,  avg # of iterations =  2.0

     Threshold (ethr) on eigenvalues was too large:
     Diagonalizing with lowered threshold

     Davidson diagonalization with overlap
     ethr =  5.50E-04,  avg # of iterations =  2.1

     total cpu time spent up to now is        1.6 secs

     total energy              =     -16.92315441 Ry
     estimated scf accuracy    <       0.05198112 Ry

     iteration #  2     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  6.50E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is        2.0 secs

     total energy              =     -16.92460727 Ry
     estimated scf accuracy    <       0.00287797 Ry

     iteration #  3     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  3.60E-05,  avg # of iterations =  2.3

     total cpu time spent up to now is        2.6 secs

     total energy              =     -16.92519518 Ry
     estimated scf accuracy    <       0.00005993 Ry

     iteration #  4     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  7.49E-07,  avg # of iterations =  5.0

     total cpu time spent up to now is        3.5 secs

     total energy              =     -16.92527313 Ry
     estimated scf accuracy    <       0.00000025 Ry

     iteration #  5     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  3.06E-09,  avg # of iterations =  6.5

     total cpu time spent up to now is        4.5 secs

     End of self-consistent calculation

     Number of k-points >= 100: set verbosity='high' to print the bands.

     highest occupied level (ev):     6.0170

!    total energy              =     -16.92527464 Ry
     estimated scf accuracy    <          9.2E-09 Ry

     The total energy is the sum of the following terms:
     one-electron contribution =       4.80531875 Ry
     hartree contribution      =       1.11313728 Ry
     xc contribution           =      -6.16103396 Ry
     ewald contribution        =     -16.68269672 Ry

     convergence has been achieved in   5 iterations

     Forces acting on atoms (cartesian axes, Ry/au):

     atom    1 type  1   force =     0.00000000    0.00000000    0.00000000
     atom    2 type  1   force =     0.00000000    0.00000000    0.00000000

     Total force =     0.000000     Total SCF correction =     0.000000


     Computing stress (Cartesian axis) and pressure

          total   stress  (Ry/bohr**3)                   (kbar)     P=        0.53
   0.00000359   0.00000000   0.00000000            0.53        0.00        0.00
   0.00000000   0.00000359  -0.00000000            0.00        0.53       -0.00
   0.00000000  -0.00000000   0.00000359            0.00       -0.00        0.53


     Writing all to output data dir ./tmp/si.save/ :
     XML data file, charge density, pseudopotentials, collected wavefunctions

     init_run     :      0.39s CPU      0.38s WALL (       1 calls)
     electrons    :      4.14s CPU      4.12s WALL (       1 calls)
     forces       :      0.08s CPU      0.07s WALL (       1 calls)
     stress       :      0.35s CPU      0.34s WALL (       1 calls)

     Called by init_run:
     wfcinit      :      0.35s CPU      0.34s WALL (       1 calls)
     potinit      :      0.01s CPU      0.01s WALL (       1 calls)
     hinit0       :      0.01s CPU      0.01s WALL (       1 calls)

     Called by electrons:
     c_bands      :      3.66s CPU      3.63s WALL (       6 calls)
     sum_band     :      0.45s CPU      0.46s WALL (       6 calls)
     v_of_rho     :      0.03s CPU      0.03s WALL (       6 calls)
     mix_rho      :      0.00s CPU      0.00s WALL (       6 calls)

     Called by c_bands:
     init_us_2    :      0.19s CPU      0.19s WALL (    1800 calls)
     cegterg      :      3.24s CPU      3.22s WALL (     720 calls)

     Called by *egterg:
     cdiaghg      :      0.05s CPU      0.05s WALL (    2865 calls)
     h_psi        :      3.27s CPU      3.23s WALL (    3105 calls)
     g_psi        :      0.01s CPU      0.01s WALL (    2265 calls)

     Called by h_psi:
     h_psi:calbec :      0.25s CPU      0.24s WALL (    3105 calls)
     vloc_psi     :      2.81s CPU      2.78s WALL (    3105 calls)
     add_vuspsi   :      0.20s CPU      0.20s WALL (    3105 calls)

     General routines
     calbec       :      0.43s CPU      0.42s WALL (    5145 calls)
     fft          :      0.02s CPU      0.02s WALL (      85 calls)
     ffts         :      0.00s CPU      0.00s WALL (       6 calls)
     fftw         :      2.92s CPU      2.90s WALL (   24674 calls)

     Parallel routines

     PWSCF        :      5.01s CPU      4.96s WALL


   This run was terminated on:  14:56:53  17Jul2025            

=------------------------------------------------------------------------------=
   JOB DONE.
=------------------------------------------------------------------------------=
