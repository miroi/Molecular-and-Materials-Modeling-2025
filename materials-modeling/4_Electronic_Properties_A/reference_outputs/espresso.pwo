
     Program PWSCF v.7.4.1 starts on 22Jul2025 at 10:33:21 

     This program is part of the open-source Quantum ESPRESSO suite
     for quantum simulation of materials; please cite
         "P. Giannozzi et al., J. Phys.:Condens. Matter 21 395502 (2009);
         "P. Giannozzi et al., J. Phys.:Condens. Matter 29 465901 (2017);
         "P. Giannozzi et al., J. Chem. Phys. 152 154105 (2020);
          URL http://www.quantum-espresso.org", 
     in publications or presentations arising from this work. More details at
     http://www.quantum-espresso.org/quote

     Parallel version (MPI), running on     4 processors

     MPI processes distributed on     1 nodes
     14889 MiB available memory on the printing compute node when the environment starts

     Reading input from espresso.pwi
Warning: card &CELL ignored
Warning: card / ignored
Warning: card &FCP ignored
Warning: card / ignored
Warning: card &RISM ignored
Warning: card / ignored

     Current dimensions of program PWSCF are:
     Max number of different atomic species (ntypx) = 10
     Max number of k-points (npk) =  40000
     Max angular momentum in pseudopotentials (lmaxx) =  4
     Message from routine setup:
     using ibrav=0 with symmetry is DISCOURAGED, use correct ibrav instead

     R & G space division:  proc/nbgrp/npool/nimage =       4
     Subspace diagonalization in iterative solution of the eigenvalue problem:
     a serial algorithm will be used


     Parallelization info
     --------------------
     sticks:   dense  smooth     PW     G-vecs:    dense   smooth      PW
     Min         238     238     73                 4892     4892     821
     Max         239     239     74                 4896     4896     823
     Sum         955     955    295                19579    19579    3287

     Using Slab Decomposition



     bravais-lattice index     =            0
     lattice parameter (alat)  =       7.3063  a.u.
     unit-cell volume          =     275.7930 (a.u.)^3
     number of atoms/cell      =            2
     number of atomic types    =            1
     number of electrons       =         8.00
     number of Kohn-Sham states=            8
     kinetic-energy cutoff     =      65.0000  Ry
     charge density cutoff     =     260.0000  Ry
     scf convergence threshold =      1.0E-10
     mixing beta               =       0.7000
     number of iterations used =            8  plain     mixing
     Exchange-correlation= PBE
                           (   1   4   3   4   0   0   0)

     celldm(1)=   7.306332  celldm(2)=   0.000000  celldm(3)=   0.000000
     celldm(4)=   0.000000  celldm(5)=   0.000000  celldm(6)=   0.000000

     crystal axes: (cart. coord. in units of alat)
               a(1) = (  -0.000000   0.707107   0.707107 )  
               a(2) = (   0.707107   0.000000   0.707107 )  
               a(3) = (   0.707107   0.707107   0.000000 )  

     reciprocal axes: (cart. coord. in units 2 pi/alat)
               b(1) = ( -0.707107  0.707107  0.707107 )  
               b(2) = (  0.707107 -0.707107  0.707107 )  
               b(3) = (  0.707107  0.707107 -0.707107 )  


     PseudoPot. # 1 for Si read from file:
     ./Si.upf
     MD5 check sum: 02fab3f35e82123ef5bf1cb05d5b1a5e
     Pseudo is Norm-conserving + core correction, Zval =  4.0
     Generated using ONCVPSP code by D. R. Hamann
     Using radial grid of 1510 points,  6 beta functions with: 
                l(1) =   0
                l(2) =   0
                l(3) =   1
                l(4) =   1
                l(5) =   2
                l(6) =   2

     atomic species   valence    mass     pseudopotential
     Si                4.00    28.08500     Si( 1.00)

     48 Sym. Ops., with inversion, found (24 have fractional translation)



   Cartesian axes

     site n.     atom                  positions (alat units)
         1        Si     tau(   1) = (   0.3535534   1.0606602   1.0606602  )
         2        Si     tau(   2) = (   0.0000000   0.0000000   0.0000000  )

     number of k points=   120  Gaussian smearing, width (Ry)=  0.0100

     Number of k-points >= 100: set verbosity='high' to print them.

     Dense  grid:    19579 G-vectors     FFT dimensions: (  40,  40,  40)

     Estimated max dynamical RAM per process >       3.96 MB

     Estimated total dynamical RAM >      15.83 MB

     Initial potential from superposition of free atoms

     starting charge       7.9988, renormalised to       8.0000
     Starting wfcs are    8 randomized atomic wfcs

     total cpu time spent up to now is        0.4 secs

     Self-consistent Calculation

     iteration #  1     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.00E-02,  avg # of iterations =  4.8

     Threshold (ethr) on eigenvalues was too large:
     Diagonalizing with lowered threshold

     Davidson diagonalization with overlap
     ethr =  6.32E-04,  avg # of iterations =  1.8

     total cpu time spent up to now is        2.7 secs

     total energy              =     -16.92303912 Ry
     estimated scf accuracy    <       0.05403840 Ry

     iteration #  2     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  6.75E-04,  avg # of iterations =  1.1

     total cpu time spent up to now is        3.6 secs

     total energy              =     -16.92515571 Ry
     estimated scf accuracy    <       0.00218582 Ry

     iteration #  3     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  2.73E-05,  avg # of iterations =  4.6

     total cpu time spent up to now is        4.7 secs

     total energy              =     -16.92526525 Ry
     estimated scf accuracy    <       0.00003735 Ry

     iteration #  4     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  4.67E-07,  avg # of iterations =  3.8

     total cpu time spent up to now is        5.9 secs

     total energy              =     -16.92527434 Ry
     estimated scf accuracy    <       0.00000016 Ry

     iteration #  5     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  2.04E-09,  avg # of iterations =  3.8

     total cpu time spent up to now is        7.1 secs

     total energy              =     -16.92527466 Ry
     estimated scf accuracy    <          9.3E-09 Ry

     iteration #  6     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.16E-10,  avg # of iterations =  2.0

     total cpu time spent up to now is        8.2 secs

     total energy              =     -16.92527466 Ry
     estimated scf accuracy    <          2.7E-10 Ry

     iteration #  7     ecut=    65.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  3.41E-12,  avg # of iterations =  2.7

     total cpu time spent up to now is        9.2 secs

     End of self-consistent calculation

     Number of k-points >= 100: set verbosity='high' to print the bands.

     the Fermi energy is     6.3158 ev

!    total energy              =     -16.92527466 Ry
     estimated scf accuracy    <          3.8E-11 Ry
     smearing contrib. (-TS)   =      -0.00000012 Ry
     internal energy E=F+TS    =     -16.92527454 Ry

     The total energy is F=E-TS. E is the sum of the following terms:
     one-electron contribution =       4.80530987 Ry
     hartree contribution      =       1.11315426 Ry
     xc contribution           =      -6.16104196 Ry
     ewald contribution        =     -16.68269672 Ry

     convergence has been achieved in   7 iterations

     Forces acting on atoms (cartesian axes, Ry/au):

     atom    1 type  1   force =     0.00000000   -0.00000000    0.00000000
     atom    2 type  1   force =     0.00000000    0.00000000    0.00000000

     Total force =     0.000000     Total SCF correction =     0.000000


     Computing stress (Cartesian axis) and pressure

          total   stress  (Ry/bohr**3)                   (kbar)     P=        0.54
   0.00000365  -0.00000000  -0.00000000            0.54       -0.00       -0.00
  -0.00000000   0.00000365  -0.00000000           -0.00        0.54       -0.00
  -0.00000000  -0.00000000   0.00000365           -0.00       -0.00        0.54


     Writing all to output data dir ./tmp/Si_Electronic_A.save/ :
     XML data file, charge density, pseudopotentials, collected wavefunctions

     init_run     :      0.36s CPU      0.38s WALL (       1 calls)
     electrons    :      8.78s CPU      8.81s WALL (       1 calls)
     forces       :      0.11s CPU      0.11s WALL (       1 calls)
     stress       :      0.48s CPU      0.47s WALL (       1 calls)

     Called by init_run:
     wfcinit      :      0.32s CPU      0.34s WALL (       1 calls)
     potinit      :      0.01s CPU      0.01s WALL (       1 calls)
     hinit0       :      0.01s CPU      0.01s WALL (       1 calls)

     Called by electrons:
     c_bands      :      7.63s CPU      7.63s WALL (       8 calls)
     sum_band     :      1.10s CPU      1.13s WALL (       8 calls)
     v_of_rho     :      0.04s CPU      0.04s WALL (       8 calls)
     mix_rho      :      0.01s CPU      0.01s WALL (       8 calls)

     Called by c_bands:
     init_us_2    :      0.24s CPU      0.24s WALL (    2280 calls)
     cegterg      :      7.05s CPU      7.05s WALL (     960 calls)

     Called by *egterg:
     cdiaghg      :      0.17s CPU      0.17s WALL (    3783 calls)
     h_psi        :      6.46s CPU      6.47s WALL (    4023 calls)
     g_psi        :      0.02s CPU      0.02s WALL (    2943 calls)

     Called by h_psi:
     h_psi:calbec :      0.48s CPU      0.48s WALL (    4023 calls)
     vloc_psi     :      5.56s CPU      5.58s WALL (    4023 calls)
     add_vuspsi   :      0.39s CPU      0.39s WALL (    4023 calls)

     General routines
     calbec       :      0.83s CPU      0.82s WALL (    6063 calls)
     fft          :      0.02s CPU      0.02s WALL (     105 calls)
     ffts         :      0.00s CPU      0.00s WALL (       8 calls)
     fftw         :      6.01s CPU      6.04s WALL (   51454 calls)
     davcio       :      0.05s CPU      0.06s WALL (    3412 calls)

     Parallel routines

     PWSCF        :      9.78s CPU      9.85s WALL


   This run was terminated on:  10:33:31  22Jul2025            

=------------------------------------------------------------------------------=
   JOB DONE.
=------------------------------------------------------------------------------=
