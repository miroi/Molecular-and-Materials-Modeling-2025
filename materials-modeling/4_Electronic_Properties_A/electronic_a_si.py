#!/usr/bin/env python
from ase import Atoms
from ase.calculators.espresso import Espresso, EspressoProfile
from ase.io import write
import os
import numpy as np
import subprocess
import shutil
import sys
sys.stdout.flush()  

# ==============================================
# ASE electronic properties 
# ==============================================
#
# ==============================================
# 1. Quantum Espresso Input Parameters
# ==============================================
pseudopotentials = {
    'Si': 'Si.upf'
}

input_data = {
    'control': {
        'prefix': 'Si_Electronic_A',
        'outdir': './tmp',
        'verbosity': 'low',
        'tstress': True,
        'tprnfor': True,
        'wf_collect': True,
        'disk_io': 'high'
    },
    'system': {
        'ecutwfc': 65,
        'occupations': 'smearing',
        'smearing': 'gauss',
        'degauss': 0.01,
        'ibrav': 0,
        'nat': 2,
        'ntyp': 1
    },
    'electrons': {
        'conv_thr': 1.0e-10
    }
}

# ==============================================
# 2. Machine-specific command setup
# ==============================================
# Set QE bin directory
#qe_bin = "/home/dsen/work/bin/qe-7.4.1_serial"
qe_bin = "/home/dsen/work/bin/qe-7.4.1"

# Main QE calculation 
#pw_command = f'{qe_bin}/bin/pw.x'
pw_command = f'mpirun -np 4 {qe_bin}/bin/pw.x'

# Serial post-processing commands for fast execution
pp_command = f"{qe_bin}/bin/pp.x < pp.in > pp.out 2>&1"
dos_command = f"{qe_bin}/bin/dos.x < dos.in > dos.out 2>&1"

# Memory intensive Parallel post-processing commands (run in one node only)
#projwfc_command = f'{qe_bin}/bin/projwfc.x < projwfc.in > projwfc.out 2>&1'
projwfc_command = f'mpirun -np 4 {qe_bin}/bin/projwfc.x < projwfc.in > projwfc.out 2>&1'

profile = EspressoProfile(
    command=pw_command,
    pseudo_dir='./'
)

# ==============================================
# 3. Atomic Structure
# ==============================================
atoms = Atoms(
    symbols=['Si']*2,
    positions=[ # positions in Angstrom
        [1.3669840665567621, 4.1009521996702860, 4.1009521996702860],
        [0.0000000000000000, 0.0000000000000000, 0.0000000000000000]
    ],
    cell=[ # positions in Angstrom
        [-0.0000000000000000, 2.7339681331135242, 2.7339681331135242],
        [2.7339681331135242, 0.0000000000000000, 2.7339681331135242],
        [2.7339681331135242, 2.7339681331135242, 0.0000000000000000]
    ],
    pbc=[True, True, True]
)
# ==============================================
# 4. Calculator Configuration 
# ==============================================
calc = Espresso(
    profile=profile,
    pseudopotentials=pseudopotentials,
    input_data=input_data,
    kpts=(15,15,15)  
)
atoms.calc = calc

# ==============================================
# 6. Run SCF Calculation
# ==============================================
print("\nRunning SCF calculation...", flush=True)
os.makedirs('tmp', exist_ok=True)
total_energy = atoms.get_potential_energy()
print(f"  Total energy: {total_energy:.6f} eV", flush=True)

# ==============================================
# 7. Calculate electronic properties
# ==============================================
# setup qe_tool
def run_qe_tool(command, input_file, tool_name):
    """Run QE tool with input file and redirect output to file"""
    try:
        with open(input_file, 'r') as fin:
            subprocess.run(command, shell=True, stdin=fin, check=True)
        print(f"  {tool_name} completed successfully", flush=True)
        return True
    except subprocess.CalledProcessError as e:
        print(f"Error running {tool_name}: {e}", flush=True)
        return False

# 7.1. Charge density calculation
print("\n1. Calculating charge density...", flush=True)
with open('pp.in', 'w') as f:
    f.write(f"""&INPUTPP
    prefix = '{input_data['control']['prefix']}',
    outdir = './tmp',
    filplot = 'charge_density',
    plot_num = 0
/
&PLOT
    iflag = 3,
    output_format = 6,
    fileout = 'charge_density.cube'
/
""")
run_qe_tool(pp_command, 'pp.in', 'pp.x')

# 7.2. TDOS calculation
print("\n2. Calculating DOS...", flush=True)
with open('dos.in', 'w') as f:
    f.write(f"""&DOS
    prefix = '{input_data['control']['prefix']}',
    outdir = './tmp',
    Emin = -20, Emax = 20, DeltaE = 0.01,
    ngauss = 0,
    degauss = 0.02,
    fildos = 'total_dos.dat'
/
""")
run_qe_tool(dos_command, 'dos.in', 'dos.x')

# 7.3. PDOS calculation
print("\n3. Calculating PDOS...", flush=True)
with open('projwfc.in', 'w') as f:
    f.write(f"""&PROJWFC
    prefix = '{input_data['control']['prefix']}',
    outdir = './tmp',
    ngauss = 0,
    degauss = 0.02,
    DeltaE = 0.01,
    Emin = -20,
    Emax = 20,
    filpdos = 'pdos',
    filproj = 'projections'
/
""")
run_qe_tool(projwfc_command, 'projwfc.in', 'projwfc.x')

# 7.4. Organize and process PDOS files
print("\n4. Organizing and processing PDOS files...", flush=True)
os.makedirs('pdos_results', exist_ok=True)
for fname in os.listdir('.'):
    if fname.startswith('pdos.') or fname.startswith('projections'):
        shutil.move(fname, os.path.join('pdos_results', fname))
print("  PDOS files moved to pdos_results directory", flush=True)

# 7.5. Löwdin population analysis 
print("\n6. Extracting Löwdin charges...", flush=True)
lowdin_file = 'lowdin.out'
spilling = None

try:
    with open('projwfc.out', 'r') as f:
        lines = f.readlines()
    
    # Find and save Löwdin charges section
    with open(lowdin_file, 'w') as out_f:
        out_f.write("Löwdin Charges from projwfc.out:\n")
        out_f.write("================================\n")
        
        in_section = False
        for line in lines:
            if "Lowdin Charges:" in line:
                in_section = True
            elif "Spilling Parameter:" in line:
                spilling = float(line.split()[-1])
                out_f.write(f"\nSpilling Parameter: {spilling:.6f}\n")
                break
            
            if in_section:
                out_f.write(line)
    
    print(f"  Löwdin charges saved to {lowdin_file}", flush=True)
    if spilling is not None:
        print(f"  Spilling parameter: {spilling:.6f} (values <0.05 are good)", flush=True)
except Exception as e:
    print(f"Error extracting Löwdin charges: {str(e)}", flush=True)

print("\n=== Electronic Structure Analysis Complete ===", flush=True)
print("Generated files:", flush=True)
print("- Charge density: charge_density.cube", flush=True)
print("- Total DOS: total_dos.dat", flush=True)
print("- PDOS files: pdos_results/", flush=True)
print("- Löwdin Charges: lowdin.out", flush=True)