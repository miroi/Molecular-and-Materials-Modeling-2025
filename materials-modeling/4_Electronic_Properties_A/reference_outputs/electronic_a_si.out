
Running SCF calculation...
  Total energy: -230.280074 eV

1. Calculating charge density...
  pp.x completed successfully

2. Calculating DOS...
  dos.x completed successfully

3. Calculating PDOS...
  projwfc.x completed successfully

4. Organizing and processing PDOS files...
  PDOS files moved to pdos_results directory

6. Extracting Löwdin charges...
  Löwdin charges saved to lowdin.out
  Spilling parameter: 0.008900 (values <0.05 are good)

=== Electronic Structure Analysis Complete ===
Generated files:
- Charge density: charge_density.cube
- Total DOS: total_dos.dat
- PDOS files: pdos_results/
- Löwdin Charges: lowdin.out
